gtsfm/utils/io.py

```diff
@@ -265,15 +265,36 @@ def read_images_txt(fpath: str) -> Tuple[Optional[List[Pose3]], Optional[List[st
     # and a record of the number of reconstructed images.
     for line in lines[4::2]:
         i, qw, qx, qy, qz, tx, ty, tz, i, img_fname = line.split()
-        # Colmap provides extrinsics, so must invert
-        iRw = Rot3(float(qw), float(qx), float(qy), float(qz))
-        wTi = Pose3(iRw, np.array([tx, ty, tz], dtype=np.float64)).inverse()
+        iRw_quaternion = np.array([qw, qx, qy, qz])
+        itw = np.array([tx, ty, tz])
+
+        wTi = recover_pose_from_extrinsics(iRw_quaternion, itw)
         wTi_list.append(wTi)
         img_fnames.append(img_fname)
 
     return wTi_list, img_fnames
 
 
+def recover_pose_from_extrinsics(iRw_quaternion: np.ndarray, itw: np.ndarray) -> Pose3:
+    """Converts the quaternion and translation values into a camera pose.
+
+    Args:
+        iRw_quaternion: camera rotation extrinsics, array of shape (4,)
+        itw: camera translation extrinsics, array of shape (3,)
+
+    Returns:
+        The pose of the camera in world frame.
+    """
+    qw, qx, qy, qz = iRw_quaternion
+    tx, ty, tz = itw
+
+    # Colmap provides extrinsics, so must invert
```

Comment:
thanks for this PR, Adi. I think this was definitely useful for debugging. I think I found the previous I/O to be more readable though. So not sure I would want to check this in. @ayushbaid @akshay-krishnan what are your thoughts?

Reply:
This was actually my suggestion to debug the issue that Adi was seeing. This is more unit-testable. This should also not be in io.py, it is a transform.